extensions/kafka/src/test/java/com/hazelcast/jet/kafka/impl/WriteKafkaPTest.java

```diff
@@ -264,10 +264,14 @@ public void test_resumeTransaction() throws Exception {
         producer.commitTransaction();
 
         // verify items are visible
-        polledRecords = consumer.poll(Duration.ofSeconds(2));
         StringBuilder actualContents = new StringBuilder();
-        for (ConsumerRecord<String, String> record : polledRecords) {
-            actualContents.append(record.value()).append('\n');
+        for (int receivedCount = 0; receivedCount < 2; ) {
```

Comment:
This will loop indefinitely if the records aren't received, failing with a generic test timeout.

Wouldn't it be better to try the poll several times and eventually fail with an error message saying "didn't receive X records within 30 seconds" or similar?

Reply:
Yeah, it would be better, but I think it's enough for now. The test maybe never failed.